engine/src/main/java/org/pentaho/di/trans/steps/fileinput/text/TextFileInputUtils.java

```diff
@@ -0,0 +1,1027 @@
+        */
+
+      if ( checkPattern( sb.toString(), regex, escapeChar ) % 2 == 0 ) {
+        return new TextFileLine( sb.toString(), lineNumberInFile, null );
+      }
+
+      sline = getLine( log, reader, encodingType, fileFormatType, line );
+
+      if ( sline == null ) {
+        return new TextFileLine( sline, lineNumberInFile, null );
+      }
+
+      // Include \n between lines ignoring \r to be OS independent
+      sb.append( "\n" + sline );
+      lineNumberInFile++;
+    } while ( true );
+  }
+
+
+  public static final String getLine( LogChannelInterface log, BufferedInputStreamReader reader, EncodingType encodingType,
+      int formatNr, StringBuilder line ) throws KettleFileException {
+    int c = 0;
+    line.setLength( 0 );
+    try {
+      switch( formatNr ) {
+        case TextFileInputMeta.FILE_FORMAT_DOS:
+          while ( c >= 0 ) {
+            c = reader.read();
+
+            if ( encodingType.isReturn( c ) || encodingType.isLinefeed( c ) ) {
+              c = reader.read(); // skip \n and \r
+              if ( !encodingType.isReturn( c ) && !encodingType.isLinefeed( c ) ) {
+                // make sure its really a linefeed or cariage return
+                // raise an error this is not a DOS file
+                // so we have pulled a character from the next line
+                throw new KettleFileException( BaseMessages.getString( PKG, "TextFileInput.Log.SingleLineFound" ) );
+              }
+              return line.toString();
+            }
+            if ( c >= 0 ) {
+              line.append( (char) c );
+            }
+          }
+          break;
+        case TextFileInputMeta.FILE_FORMAT_UNIX:
+          while ( c >= 0 ) {
+            c = reader.read();
+
+            if ( encodingType.isLinefeed( c ) || encodingType.isReturn( c ) ) {
+              return line.toString();
+            }
+            if ( c >= 0 ) {
+              line.append( (char) c );
+            }
+          }
+          break;
+        case TextFileInputMeta.FILE_FORMAT_MIXED:
+          // in mixed mode we suppose that either 1) LF is the last char, or 2) CR is last character if followed by a
+          // non-linefeed character
+          while ( c >= 0 ) {
+            c = reader.read();
+
+            if ( encodingType.isLinefeed( c ) ) {
+              return line.toString();
+            } else if ( encodingType.isReturn( c ) ) {
+              if ( reader.markSupported() ) {
+                reader.mark( 1 );
+                c = reader.read();
+                if ( !encodingType.isLinefeed( c ) ) {
+                  reader.reset();
+                }
+                return line.toString();
+              }
+            } else {
+              if ( c >= 0 ) {
+                line.append( (char) c );
+              }
+            }
+          }
+          break;
+        default:
+          break;
+      }
+    } catch ( KettleFileException e ) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An exception is caught and re-thrown within the catch clause without any other operations being performed on the exception, making the catch clause unnecessary.

**Fix**: Perform additional operations on the exception that has been caught. You can wrap a standard exception into a custom exception to add additional information or special handling. When you do that, make sure to set the original exception as the cause. Otherwise, you will lose the stack trace and message of the original exception, which will make it difficult to analyze the event that caused the exception.  
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)